tools/tool-testing/galaxy-utils.js

```diff
@@ -0,0 +1,254 @@
+var selftest = require('../tool-testing/selftest.js');
+var Run = selftest.Run;
+var testUtils = require('../tool-testing/test-utils.js');
+var files = require('../fs/files.js');
+var utils = require('../utils/utils.js');
+var authClient = require('../meteor-services/auth-client.js');
+var auth = require('../meteor-services/auth.js');
+
+// Run curl with the given specifications. Return an instance of Run.
+var runCurl = function (/*args*/) {
+  var curl = "/usr/bin/curl";
+  return new Run(curl, {
+    args: _.toArray(arguments)
+  });
+};
+
+// Some constants.
+var GALAXY_USERNAME = process.env.GALAXY_USERNAME;
+var GALAXY_PASSWORD = process.env.GALAXY_PASSWORD;
+var GALAXY_URL = process.env.DEPLOY_HOSTNAME;
+var GALAXY_MOCK_MODE = process.env.GALAXY_MOCK_MODE;
+
+// Fail if the test is obviously not set up for using Galaxy.
+//
+// Make sure that we have at least set all of the variables that we need to run
+// against a Galaxy (GALAXY_URL, username & password). An extra safety
+// check to avoid strange errors, deploying/calling random methods on Mother,
+// etc.
+exports.sanityCheck = selftest.markStack(function () {
+  if (! GALAXY_URL ) {
+    selftest.fail("Please specify a GALAXY_URL to test against Galaxy.\n");
+  }
+  if (! GALAXY_USERNAME ||
+      ! GALAXY_PASSWORD) {
+    selftest.fail(
+      "Can't use test account with Galaxy. " +
+       "Please specify GALAXY_USERNAME and GALAXY_PASSWORD.\n");
+  }
+  if (! process.env.APP_MONGO) {
+    selftest.fail(
+      "Please provide an APP_MONGO url to use for deployed apps.\n");
+   }
+});
+
+// Login to Galaxy with environment-variable credentials passed in by the user.
+//
+// Unlike the normal `meteor deploy` Galaxy is not yet publically available, so
+// we don't want to use the publically-accessible test account here.
+exports.loginToGalaxy = selftest.markStack(function (sandbox) {
+  var user = GALAXY_USERNAME;
+  var pass = GALAXY_PASSWORD;
+  testUtils.login(sandbox, user, pass);
+});
+
+// Curl an app running on Galaxy. Automatically follow redirects.
+//
+// Dealing with DNS on Galaxy can be complicated. The standard way to ensure
+// that we hit the right app on the right Galaxy is to curl the galaxy origin,
+// with the host header set to our query app.
+exports.curlToGalaxy = selftest.markStack(function (url) {
+  var hostHeader = "host: " + url;
+  var galaxyOrigin = GALAXY_URL;
+  return runCurl("-vLH", hostHeader, galaxyOrigin);
+});
+
+// String we expect to hit on 200 OK.
+exports.httpOK = "HTTP/1.1 200 OK";
+
+// String we expect to hit when using SSL.
+exports.httpRedirect = "HTTP/1.1 307 Temporary Redirect";
+
+// We expect containers to take some time to startup.
+//
+// In the future, we can use this function to poll whether the containers have
+// started and maybe our tests will be faster.
+exports.waitForContainers = selftest.markStack(function () {
+  // We are not spinning up any containers in mock mode, so don't wait too long.
```

Comment:
This isn't strictly speaking true. The mocked scheduler does "spin up containers" (ie it takes a little while for container records to appear). I'm not sure we care though in these tests but I thought I'd mention it.